clients/src/main/java/org/apache/kafka/clients/consumer/internals/HeartbeatRequestManager.java

```diff
@@ -188,9 +188,7 @@ private NetworkClientDelegate.UnsentRequest makeHeartbeatRequest() {
             if (response != null) {
                 onResponse((ConsumerGroupHeartbeatResponse) response.responseBody(), response.receivedTimeMs());
```

Comment:
Thanks
`request.handler().completionTimeMs() instead of response.receivedTimeMs()` - This is addressed

`UnsentRequest using request.handler() would not be possible` - Let me quickly follow up with a MINOR PR to address this comment.